src/languages/swift.js

```diff
@@ -126,22 +127,98 @@ export default function(hljs) {
       SINGLE_LINE_STRING("###"),
     ]
   };
+
+  // https://docs.swift.org/swift-book/ReferenceManual/LexicalStructure.html#ID412
+  const QUOTED_IDENTIFIER = {
+    begin: concat(/`/, Swift.identifier, /`/)
+  };
+  const IMPLICIT_PARAMETER = {
+    className: 'variable',
+    begin: /\$\d+/
+  };
+  const PROPERTY_WRAPPER_PROJECTION = {
+    className: 'variable',
+    begin: `\\$${Swift.identifierCharacter}+`
+  };
+  const IDENTIFIERS = [QUOTED_IDENTIFIER, IMPLICIT_PARAMETER, PROPERTY_WRAPPER_PROJECTION];
+
+  // https://docs.swift.org/swift-book/ReferenceManual/Attributes.html
+  const AVAILABLE = {
+    begin: /(@|#)available/,
+    returnBegin: true,
+    contains: [
+      {
+        className: 'keyword',
+        begin: /(@|#)available/
+      },
+      {
+        begin: /\(/,
+        end: /\)/,
+        endsParent: true,
```

Comment:
Necessary?

Reply:
Do you mean the `endsParent`? I wasn't sure.

Will the parent rule automatically end if there's no more matching `contains` rule?
Will `endsParent` speed up the matching? (so it doesn't have to look for another subrule)

Reply:
> Will the parent rule automatically end if there's no more matching contains rule?

Yes, if there is no `end` matcher to "hold the mode open".

> Will endsParent speed up the matching? (so it doesn't have to look for another subrule)

*Technically*, but not to the extent it should matter and we don't use this type of thinking in any other grammars.  If it's clear (gramatically) that a rule will short-circuit itself, we usually just let it.

Reply:
Done (in the next commit).